content/en/blog/_posts/2024-05-13-cloud-provider-migration.md

```diff
@@ -0,0 +1,86 @@
+---
+layout: blog
+title: 'Completing the largest migration in Kubernetes history'
+date: 2024-05-20
+slug: completing-cloud-provider-migration
+author: >
+  Andrew Sy Kim (Google),
+  Michelle Au (Google),
+  Michael McCune (Red Hat)
+---
+
+Since as early as Kubernetes v1.7, the Kubernetes project has pursued the ambitious goal of removing built-in cloud provider integrations.
+While these integrations were instrumental in Kubernetes' early development and growth, their removal was driven by two key factors:
+the growing complexity of maintaining native support for every cloud provider across millions of lines of Go code, and the desire to establish
+Kubernetes as a truly vendor-neutral platform.
+
+After many releases, we're thrilled to announce that all cloud provider integrations have been successfully migrated from the core Kubernetes repository to external plugins.
+This migration was a complex and long-running effort due to the numerous impacted components and the critical code paths that relied on the built-in integrations for the
+five initial cloud providers: Google Cloud, AWS, Azure, OpenStack, and vSphere. To successfully complete this migration, we had to build four new subsystems from the ground up:
+
+1. [Cloud Controller Manager](https://kubernetes.io/docs/concepts/architecture/cloud-controller/)
+2. [API Server Network Proxy](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/1281-network-proxy)
+3. [Kubelet Credential Provider Plugins](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2133-kubelet-credential-providers)
+4. [CSI In-tree Storage migration plugins](https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/625-csi-migration/README.md)
+
+Each subsystem was critical to achieve full feature parity with built-in capabilities and required several releases to bring each subsystem to GA-level maturity with a safe and
+reliable migration path. More on each subsystem below.
+
+### Cloud Controller Manager
+
+The cloud controller manager was the first external component introduced in this effort, replacing functionality within the kube-controller-manager and kubelet that directly interacted with cloud APIs.
+This essential component is responsible for initializing nodes by applying metadata labels that indicate the cloud region and zone a Node is running on, as well as IP addresses that are only known to the cloud provider.
+Additionally, it runs the service controller, which is responsible for provisioning cloud load balancers for Services of type LoadBalancer.
+
+![Kubernetes components](/images/docs/components-of-kubernetes.svg)
+
+Learn more about Cloud Controller Manager [here](/docs/concepts/architecture/cloud-controller).
+
+### API Server Network Proxy
+
+The API Server Network Proxy project, initiated in 2018 in collaboration with SIG API Machinery, aimed to replace the SSH tunneler functionality within the kube-apiserver.
```

Comment:
(nit)
```suggestion
The _API server network proxy_ project, initiated in 2018 in collaboration with SIG API Machinery, aimed to replace the existing SSH tunneling functionality within the kube-apiserver.
```

I think “existing” helps clarity, but it's up to you if you don't like the tautology.